### DISCUSSION TITLE
How to add script tags in nextjs

### DISCUSSION THREAD

**Discussion by nitink66** [vercel/next.js#16927] [Help] [answered]
I'm trying to place ads on my website built with nextjs, this is the code i want to insert in my '_document.js' file
```
<script type="text/javascript">
	atOptions = {
		'key' : '2ddba543f4dc805c035ab0bf248fd',
		'format' : 'iframe',
		'height' : 300,
		'width' : 160,
		'params' : {}
	};
	document.write('<scr' + 'ipt type="text/javascript" src="http' + (location.protocol === 'https:' ? 's' : '') + '://www.displaynetworkprofit.com/2ddba543f4dc805c035ab0bf248fd/invoke.js"></scr' + 'ipt>');
</script>
```

But if i have the same , then error document is not defined throws up,Any help please

**Comment by jamesmosier** [comment, marked as answer]
You could wrap your code in an `if` statement:

```js
if (typeof document !== 'undefined') {
  atOptions = {
    'key' : '2ddba543f4dc805c035ab0bf248fd',
    'format' : 'iframe',
    'height' : 300,
    'width' : 160,
    'params' : {}
  };
  document.write('<scr' + 'ipt type="text/javascript" src="http' + (location.protocol === 'https:' ? 's' : '') + '://www.displaynetworkprofit.com/2ddba543f4dc805c035ab0bf248fd/invoke.js"></scr' + 'ipt>');
}
```

Alternatively the [with-google-analytics](https://github.com/vercel/next.js/blob/canary/examples/with-google-analytics/pages/_document.js) example does it like this:
```js
<script
  dangerouslySetInnerHTML={{
    __html: `
  atOptions = {
    'key' : '2ddba543f4dc805c035ab0bf248fd',
    'format' : 'iframe',
    'height' : 300,
    'width' : 160,
    'params' : {}
  };
  document.write('<scr' + 'ipt type="text/javascript" src="http' + (location.protocol === 'https:' ? 's' : '') + '://www.displaynetworkprofit.com/2ddba543f4dc805c035ab0bf248fd/invoke.js"></scr' + 'ipt>');
`,
  }}
/>
```

**Reply by mariohmol** [reply]
Is it possible to insert in a specific page and not in the head? most of the ads includes the banner where u place the script tag, so if u include on head the ads will appear on top of the page

**Reply by jamesmosier** [reply]
Yes, check out the new `Script` component: https://nextjs.org/docs/basic-features/script

**Reply by mariohmol** [reply]
I see, in my understanding this script includes the tag at the head of the html, and not in the body... am I missing some specific configuration?

**Comment by AneelaBrister** [comment]
Hi there, question for you -- is this the AdsTerra platform for ads? Dealing with a different situation with this.
Thanks.
